src/main/starlark/builtins_bzl/common/cc/experimental_cc_static_library.bzl

```diff
@@ -0,0 +1,301 @@
+        variables = archiver_variables,
+    )
+    args = actions.args()
+    args.add_all(command_line)
+    args.add_all(objects)
+
+    if cc_common.is_enabled(
+        feature_configuration = feature_configuration,
+        feature_name = "archive_param_file",
+    ):
+        args.use_param_file("@%s", use_always = True)
+
+    env = cc_common.get_environment_variables(
+        feature_configuration = feature_configuration,
+        action_name = ACTION_NAMES.cpp_link_static_library,
+        variables = archiver_variables,
+    )
+    execution_requirements_keys = cc_common.get_execution_requirements(
+        feature_configuration = feature_configuration,
+        action_name = ACTION_NAMES.cpp_link_static_library,
+    )
+
+    actions.run(
+        executable = archiver_path,
+        arguments = [args],
+        env = env,
+        execution_requirements = {k: "" for k in execution_requirements_keys},
+        inputs = depset(transitive = [cc_toolchain.all_files, objects]),
+        outputs = [static_library],
+        use_default_shell_env = True,
+        mnemonic = "CppTransitiveArchive",
+        progress_message = "Creating static library %{output}",
+    )
+
+    return static_library
+
+def _validate_static_library(*, name, actions, cc_toolchain, feature_configuration, static_library):
+    if not cc_common.action_is_enabled(
+        feature_configuration = feature_configuration,
+        action_name = ACTION_NAMES.validate_static_library,
+    ):
+        return None
+
+    validation_output = actions.declare_file(name + "_validation_output.txt")
+
+    validator_path = cc_common.get_tool_for_action(
+        feature_configuration = feature_configuration,
+        action_name = ACTION_NAMES.validate_static_library,
+    )
+    args = actions.args()
+    args.add(static_library)
+    args.add(validation_output)
+
+    execution_requirements_keys = cc_common.get_execution_requirements(
+        feature_configuration = feature_configuration,
+        action_name = ACTION_NAMES.validate_static_library,
+    )
+
+    actions.run(
+        executable = validator_path,
+        arguments = [args],
+        execution_requirements = {k: "" for k in execution_requirements_keys},
+        inputs = depset(
+            direct = [static_library],
+            transitive = [cc_toolchain.all_files],
+        ),
+        outputs = [validation_output],
+        use_default_shell_env = True,
+        mnemonic = "ValidateStaticLibrary",
+        progress_message = "Validating static library %{label}",
+    )
+
+    return validation_output
+
+def _pretty_label(label):
+    s = str(label)
+
+    # Emit main repo labels (both with and without --enable_bzlmod) without a
+    # repo prefix.
+    if s.startswith("@@//") or s.startswith("@//"):
+        return s.lstrip("@")
+    return s
+
+def _linkdeps_map_each(linker_input):
+    has_library = False
+    for lib in linker_input.libraries:
+        if lib.pic_objects or lib.objects:
```

Comment:
Keep the condition consistent with the one mentioned in a comment above. I.e. `objects != None`